### ISSUE TITLE
 MODS package manager show a blank page XPEnology DSM6.2.2 DS918+ 

### ISSUE THREAD

**Issue by terryche** [vletroye/SynoPackages#4]
Hi, vletroye,
I'm using a XPEnology DSM 6.2.2 DS918+. CPU A4-5000.
When I was using DSM 6.1.7, MODS package manager displayed nothing. Then I upgraded to 6.22. But it's still a blank page. Is this becaus of the AMD CPU?

Thanks,
Terry

**Comment by vletroye** [comment]
Another user reports the same issue with a DS1815+ (same model as mine) : a white page.
He also reports that he get a popup with the message : "The web page at https://<his Synology>/webman/3rdparty/MODS_PackageManager/mvpkg.php might be temporarily down or it may have moved permanently to a new web address."

I have so far no idea why this happens. Really not easy to debug "remotely".

But the CPU should have no influence. It's purely php and shell scripts.

**Comment by vletroye** [comment]
Only one thing I could think about... A timeout. If a shell script take too much time.

Looking inot the DevTool Console of Chrome (F12), you coud see messages like "Unchecked runtime.lastError: The message port closed before a response was received." mvpkg.php:1

**Comment by vletroye** [comment]
And there is a test you can do if you can connect on your NAS via SSH.

Open a SSH console (Putty on Windows, GateOne or Web Console package on your NAS)

In that console, type: php /var/packages/MODS_PackageManager/target/ui/mvpkg.php
You should get the html of the page with plenty of <option data-volume='volumeX' value='some package'>volumeX

If it does not work, try to execute it as 'root':
1. Type the command and give your root account (same as NAS administrator) : sudo -i
2. type the command above.

**Comment by Presjar** [comment]
Hi Veltroye, thanks for the follow up email and this thread.

I completed the test you listed, running the "php /var/packages/MODS_PackageManager/target/ui/mvpkg.php"

It required my admin password to run, but it did then print the HTML to the console.

**Comment by vletroye** [comment]
Ok. The php script works fine... One can see your packages in the html

<option data-volume='volume1' value='CloudSync'>volume1
<option data-volume='volume1' value='Docker'>volume1
<option data-volume='volume1' value='DownloadStation'>volume1
<option data-volume='volume1' value='MariaDB'>volume1
<option data-volume='volume1' value='MODS_PackageManager'>volume1
<option data-volume='volume1' value='mono'>volume1
<option data-volume='volume1' value='Node.js_v4'>volume1
<option data-volume='volume1' value='nzbdrone'>volume1
<option data-volume='volume1' value='PHP5.6'>volume1
<option data-volume='volume1' value='PHP7.0'>volume1
<option data-volume='volume1' value='py3k'>volume1
<option data-volume='volume1' value='python'>volume1
<option data-volume='volume1' value='sabnzbd'>volume1
<option data-volume='volume1' value='Spotweb'>volume1
<option data-volume='volume1' value='SynologyApplicationService'>volume1
<option data-volume='volume1' value='WebDAVServer'>volume1
<option data-volume='volume1' value='WebStation'>volume1

It's normal (I think) that you had to type your password. The package doesn't need it has it runs as admin.... (Be to be honest, I am not at all expert in that)

The problem is possibly with your webserver. Are you using apache or nginx ? My package needs nginx.
And, is php 7.0 installed ? My package needs that version.

**Comment by Presjar** [comment]
Ok. I have apache installed and selected in the webstation app. Also php7
is installed but I also have another php version installed also.

I'll stop webstation, removed apache and the other php version. Can't right
now as the webstation is in use. Will report back asap.

**Comment by Presjar** [comment]
I removed the PHP 5.6 and apache applications and restarted Web Station and
the MODS package.
No success, still the white screen error with the URL.

Perhaps I have conflict with another application.

**Comment by vletroye** [comment]
So now you are using nginx in your WebStation with PHP7 by default, and it still does not display the page? I don't see which conflict there could be. It's a pure php script executing shell commands...

Possibly try to open your DSM in http instead of https ? I don't except this make any difference... But as I have no smarter idea... :/

NB.: I didn't uninstall apache and older versions of PHP from my NAS. But nginx and PHP7 are defined as defaults in WebStation.

**Comment by Presjar** [comment]
NGIX and PHP 7 yes.

When I connect without https the package manager page is now loading fine!
So I can work around and move the apps I need and then go back to https.

I run my own https certificate from let's encrypt usually.

**Comment by vletroye** [comment]
I am using https also from time to time, but with the Synology certificate. No idea why this would make a difference...

**Comment by aschip** [comment]
This is going to sound crazy but this worked for me.

I am using DS718+, I have installed "Php7","Web Station", "Init 3rd Party" and MODS_PackageManager".
Then made sure that I used nginx in web station.

But still, I got this error when I opened it:
![image](https://user-images.githubusercontent.com/13371053/59969436-e91feb80-95a0-11e9-90d1-6a8aead18c81.png)

So I copied that url and tried it in Chrome -> new tab:

![image](https://user-images.githubusercontent.com/13371053/59969451-1b314d80-95a1-11e9-9346-bb8bf846f04c.png)

The error code was "ERR_SPDY_PROTOCOL_ERROR", so I tried all solutions mentioned here: https://zenguard.zendesk.com/hc/en-us/articles/204423722-Chrome-error-ERR-SPDY-PROTOCOL-ERROR-
 ... didn't help much ! The error persisted

So I decided to use Microsoft Edge instead of chrome & 🎉  it works!! 🤷‍♂
